modules/alloy/src/main/ts/ephox/alloy/positioning/layout/PinnedLayout.ts

```diff
@@ -0,0 +1,41 @@
+import { nu as NuSpotInfo } from '../view/SpotInfo';
+import { Bubble } from './Bubble';
+import * as Direction from './Direction';
+import { boundsRestriction } from './LayoutBounds';
+import { AnchorBox, AnchorElement, AnchorLayout } from './LayoutTypes';
+
+const pinAtTop: AnchorLayout = (
```

Comment:
NOTE: there is nothing inherently "pinning" about these layouts. They just use the anchor box top and anchor box bottom, and don't really care about anything else. The "pinning" comes from the fact that in Bounder, there is limitX and limitY code which ensures that the x and y values are within the bounds. So in combination, this "docks" the toolbar to the top or the bottom.